CMakeLists.txt

```diff
@@ -70,6 +70,23 @@ endif()
 
 # message(STATUS "BLAS Libraries: ${BLAS_LIBRARIES}")
 
+option(ENABLE_COVERAGE "Enable coverage" OFF)
+
+if (ENABLE_COVERAGE)
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+        set(COVERAGE_FLAGS "-fprofile-arcs -ftest-coverage")
```

Comment:
Should this be `add_compile_options("-fprofile-arcs" "-ftest-coverage")` (or `target_options`). See [this](https://cmake.org/cmake/help/latest/command/add_compile_options.html)
 
https://github.com/Microsoft/rDSN/blob/master/CMakeLists.txt#L31-L34